pkg/apis/core/validation/validation.go

```diff
@@ -4214,6 +4218,11 @@ func ValidatePodSpec(spec *core.PodSpec, podMeta *metav1.ObjectMeta, fldPath *fi
 	allErrs = append(allErrs, validateContainers(spec.Containers, vols, podClaimNames, gracePeriod, fldPath.Child("containers"), opts, &spec.RestartPolicy, hostUsers)...)
 	allErrs = append(allErrs, validateInitContainers(spec.InitContainers, spec.Containers, vols, podClaimNames, gracePeriod, fldPath.Child("initContainers"), opts, &spec.RestartPolicy, hostUsers)...)
 	allErrs = append(allErrs, validateEphemeralContainers(spec.EphemeralContainers, spec.Containers, spec.InitContainers, vols, podClaimNames, fldPath.Child("ephemeralContainers"), opts, &spec.RestartPolicy, hostUsers)...)
+
+	if opts.PodLevelResourcesEnabled {
+		allErrs = append(allErrs, validatePodResources(spec, podClaimNames, fldPath.Child("resources"), opts)...)
```

Comment:
IIUC `opts.PodLevelResourcesEnabled` is equal to `utilfeature.DefaultFeatureGate.Enabled(features.PodLevelResources) ` here.

@msau42 Is it possible for the field to be set when the feature gate is disabled? If so, we will need to reject create requests. Also does ratcheting validation apply here?

Reply:
@kerthcet Ah I missed it. Thanks for pointing that out!

Reply:
Right - it IS possible to arrive here with the gate disabled AND a value set, so we always use this pattern in validation.